endpoints/internal-check-file-acl.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Automattic\VIP\Files\Acl;
+
+require_once __DIR__ . '/../files/acl/pre-wp-utils.php';
+
+$file_request_uri = $_SERVER['HTTP_X_ORIGINAL_URI'] ?? null;
+
+if ( ! $file_request_uri ) {
+	trigger_error( 'VIP Files ACL failed due to empty URI', E_USER_WARNING );
+
+	http_response_code( 500 );
+
+	exit;
+}
+
+$file_path = parse_url( $file_request_uri, PHP_URL_PATH );
+
+$is_valid_path = Pre_WP_Utils\validate_path( $file_path );
+if ( ! $is_valid_path ) {
+	http_response_code( 500 );
+
+	exit;
+}
+
+list( $subdirectory, $sanitized_file_path ) = Pre_WP_Utils\sanitize_and_split_path( $file_path );
+
+if ( $subdirectory ) {
+	$_SERVER['REQUEST_URI'] = $subdirectory . ( $_SERVER['REQUEST_URI'] ?? '' );
+}
+
+// Unset vars we no longer need, so they don't leak into global scope.
+// Should probably move the logic above into a function :)
```

Comment:
+1 on encapsulating that into a function. Not only removes the potential for global pollution, but makes the setup testable.

Reply:
Done in e5c8b01.

It felt awkward since it's now a function that does multiple things but I agree that this is a bit cleaner and safer.